src/react/components/pages/projectSettings/projectSettingsPage.tsx

```diff
@@ -68,13 +68,24 @@ export default class ProjectSettingsPage extends React.Component<IProjectSetting
             ...formData,
         };
 
-        const isNew = !(!!projectToUpdate.id);
-        await this.props.actions.saveProject(projectToUpdate);
+        const projectName = formData["name"];
```

Comment:
I would recommend that we move all this logic into the `saveProject` action and/or the `projectService` that the action uses... That will keep our save logic all in one place and have it run wherever a project is saved.

It looks like there is a way to get access to the full redux state from within an action call - https://stackoverflow.com/questions/35667249/accessing-redux-state-in-an-action-creator

Let's explore this.  Then we can have the action pull the recent project list and perform the required checks.

Reply:
That makes sense. I’ll work on that today

Reply:
Also - Let's add some new tests to verify this check is working as expected.